EDU-751: Add caution blurb to StartToCloseTimeout (#2058)
temporalio/documentation: docs-src/concepts/what-is-a-start-to-close-timeout.md

**The default Start-To-Close Timeout is the same as the default [Schedule-To-Close Timeout](/concepts/what-is-a-schedule-to-close-timeout).**

An Activity Execution must have either this timeout (Start-To-Close) or the [Schedule-To-Close Timeout](/concepts/what-is-a-schedule-to-close-timeout) set.

> We recommend always setting this timeout; however, make sure that Start-To-Close Timeout is always set to be longer than the maximum possible time for the Activity Execution to take place.

Reviewer: Suggest changing "take place" to "complete".

For long running Activity Executions, we recommend also using [Activity Heartbeats](/concepts/what-is-an-activity-heartbeat) and [Heartbeat Timeouts](/concepts/what-is-a-heartbeat-timeout).

:::caution

We strongly recommend setting Start-To-Close Timeout.

> The Temporal Server does not detect failures when Worker lose communication with Server, such as crash. Therefore Temporal Server relies on this timeout to force Activity retries.

Reviewer: ```suggestion
The Temporal Server does not detect failures when a Worker loses communication with the Server or crashes.
Therefore, the Temporal Server relies on this timeout to force Activity retries.
```

**The default Start-To-Close Timeout is the same as the default [Schedule-To-Close Timeout](/concepts/what-is-a-schedule-to-close-timeout).**

An Activity Execution must have either this timeout (Start-To-Close) or the [Schedule-To-Close Timeout](/concepts/what-is-a-schedule-to-close-timeout) set.
We recommend always setting this timeout; however, make sure that Start-To-Close Timeout is always set to be longer than the maximum possible time for the Activity Execution to complete.
For long running Activity Executions, we recommend also using [Activity Heartbeats](/concepts/what-is-an-activity-heartbeat) and [Heartbeat Timeouts](/concepts/what-is-a-heartbeat-timeout).

> :::caution

Reviewer 2: Caution is so scary...

Can easily make this a happy `info` or `success` admonition.
We want it to be a good thing :)
